server/services/courtRegisterService.ts

```diff
@@ -46,15 +47,37 @@ export default class CourtRegisterService {
     return { courts: await CourtRegisterService.restClient(token).get({ path: `/courts/all` }) } as AllCourts
   }
 
-  async getPageOfCourts(context: Context, pageNumber: number, pageSize: number): Promise<CourtsPage> {
+  async getPageOfCourts(
+    context: Context,
+    pageNumber: number,
+    pageSize: number,
+    filter: AllCourtsFilter
+  ): Promise<CourtsPage> {
     const token = await this.hmppsAuthClient.getApiClientToken(context.username)
-    logger.info(`getting details for page of courts`)
+    logger.info(`getting details for page of courts with filter ${filter}`)
     return (await CourtRegisterService.restClient(token).get({
       path: `/courts/paged`,
-      query: `page=${pageNumber}&size=${pageSize}&sort=courtName`,
+      query: `page=${pageNumber}&size=${pageSize}&sort=courtName${this.filterToQueryParms(filter)}`,
     })) as CourtsPage
   }
 
+  filterToQueryParms(filter: AllCourtsFilter): string {
```

Comment:
would be better to use a library rather than trying to do manually e.g. we use https://nodejs.org/api/querystring.html in manage auth accounts and then you can call `querystring.stringify` to do it automatically